Devil's Punishment v2/Assets/Devil's Punishment Source/Scripts/Player/GunController.cs

```diff
@@ -541,20 +563,30 @@ IEnumerator Reload()
 
 		float swayLimit = 10.0f;
 
-		if (aiming < .5f && inputEnabled)
+		if (inputEnabled)
 		{
 			swayOffset = new Vector3(
-				Mathf.Clamp(swayOffset.x - Input.GetAxisRaw("Mouse X"), -swayLimit, swayLimit) + recoil.x * 2.0f,
+				Mathf.Clamp(swayOffset.x - Input.GetAxisRaw("Mouse X"), -swayLimit, swayLimit) + recoil.x,
 				Mathf.Clamp(swayOffset.y - Input.GetAxisRaw("Mouse Y"), -swayLimit, swayLimit),
 				0f);
 		}
+
+		if (isAiming)
+		{
+			recoil = Vector2.Lerp(recoil, Vector2.zero, Time.deltaTime * recoilRecAiming);

```

Comment:
`recoil` Lerp keeps on computing even after the value has reached very low upto e^-5, e^-13 etc
So you can have a bool either for the whole `if(isAiming) - else` or individual variables `recoil` and `swayOffset`
This is because the last parameter `t` of lerp doesn't change, but that's fine here, just a condition would do

Reply:
Might add a check value <= Mathf.Epsilon to stop the lerping down when close to zero.